packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,234 @@
+Group header view will extend `TableColumnTextGroupHeaderViewBase` and use its template.
+
+### States
+
+N/A
+
+### Accessibility
+
+Accessibility of the cells rendered using the `nimble-table-column-file-size` are handled via the [`nimble-table-cell`](https://github.com/ni/nimble/blob/f663c38741e731bef91aa58e8fb2d1cec653b679/packages/nimble-components/src/table/components/cell/template.ts#L6) which has a `role` of [`cell`](https://w3c.github.io/aria/#cell).
+
+### Mobile
+
+N/A
+
+### Globalization
+
+#### Localization of the number
+
+For this we will use `Intl.NumberFormat`, passing in the value of the `lang` design token as the locale.
+
+#### Localization of the unit label
+
+This is more difficult, as there does not seem to be a great API for getting a file size unit string for a given locale. The only implemenation I've found is `Intl.NumberFormat`. This function can also do the numeric conversion, but we would not use it for that, because it only supports converting by using a factor of 1000, not 1024. We could instead do the number conversion ourselves and just ask `Intl.NumberFormat` to translate the unit string we need. However, we would have to work around several limitations:
+
+-   Does not have translations for KiB/MiB/etc.
+-   Does not translate plural "bytes".
+-   Throws error for locales with lang subtag `zh_CN` and `zh_TW` -- expects just `zh`.
+-   Requires creating separate `Intl.NumberFormat` instances for each unit we wish to translate.
+-   Missing translations for "byte" in French (gives "o" instead of "octet"), Japanese ("バイト"), Chinese ("字节" in simplified or "位元組" in traditional), Korean ("바이트"), Russian ("Б" instead of "байт"), and probably others.
+
+Options:
+
+1. Find translations of each of our unit labels ("byte", "bytes", "KB", "MB", "GB", "TB", "PB", "KiB", "MiB", "GiB", "TiB", "PiB") for a fixed set of languages we wish to support. Maintain a mapping of language codes (e.g. "fr", "de", "zh_CN") to arrays of those localized unit labels. Given a locale to use, look up its language subtag in our map. If not found, fall back to English.
+
+**Pros:** No work for clients.
+
+**Cons:** Only supports a fixed set of languages. Up to us to find accurate translations. Unsure if subtags other than the language (e.g. region) could be relevant to the translation.
+
+2. Use Nimble label provider. Add label tokens for the units, and rely on clients to provide translations.
+
+**Pros:** Supports any language a client cares to provide translations for.
+
+**Cons:** Less convenient for clients. Does not honor the `lang` setting on the page or on `nimble-theme-provider`.
+
+3. Use `Intl.NumberFormat` and drop support for units based on a factor of 1024.
+
+**Pros:** No translation work for the client or Nimble team. Honors `lang` value.
+
+**Cons:** No support for 1024-based units. Shows "byte" unit even for multiple bytes. Missing several translations for "byte". Unclear if missing translations for other units (KB, MB, etc.).
```

Comment:
As mentioned above, I don't think the unit translations is an issue here, nor is the plural "bytes".